src/peft/tuners/lora.py

```diff
@@ -600,14 +606,19 @@ def add_weighted_adapter(self, adapters, weights, adapter_name, combination_type
         # if there is only one adapter, we can only use linear merging
         combination_type = "linear" if len(adapters) == 1 else combination_type
 
-        # new rank is the max of all ranks of the adapters
-        unique_ranks = list({self.peft_config[adapter].r for adapter in adapters})
+        adapters_ranks = [self.peft_config[adapter].r for adapter in adapters]
         if combination_type == "linear":
-            if len(unique_ranks) != 1:
+            # all adapters ranks should be same, new rank is just this value
+            if len(set(adapters_ranks)) != 1:
                 raise ValueError("All adapters must have the same r value when using `linear` combination_type")
-            new_rank = unique_ranks[0]
+            new_rank = adapters_ranks[0]
+        elif combination_type == "cat":
+            # adapters ranks may be different, new rank is sum of all ranks
+            # be careful, because output adapter rank may be really big if mixing a lot of adapters
```

Comment:
I think this comment is very valuable, but users typically don't read the source code. Therefore, I'd put it in the docstring of `combination_type` instead. Maybe also mention that this can lead to OOM errors, in case the implication is not clear.

Reply:
Fixed this one, updated docstring with warning of probable OOM errors.